frontend/cli/cmd_serve.go

```diff
@@ -177,18 +177,19 @@ func (s *serveCommonConfig) run(
 		controllerAddresses = append(controllerAddresses, controllerBind)
 	}
 
-	for _, addr := range controllerAddresses {
-		// Add controller address to allow origins for console requests.
-		// The console is run on `localhost` so we replace 127.0.0.1 with localhost.
-		if addr.Hostname() == "127.0.0.1" {
-			addr.Host = "localhost" + ":" + addr.Port()
-		}
-		s.CommonConfig.AllowOrigins = append(s.CommonConfig.AllowOrigins, addr)
-	}
-
-	if devMode && len(s.CommonConfig.AllowHeaders) == 0 {
-		// Default to allowing all headers for console requests in local dev mode.
+	if devMode {
+		// Default to allowing all origins and headers for console requests in local dev mode.
+		s.CommonConfig.AllowOrigins = []*url.URL{{Scheme: "*", Host: "*"}}
 		s.CommonConfig.AllowHeaders = []string{"*"}
+	} else {
+		for _, addr := range controllerAddresses {
+			// Add controller address to allow origins for console requests.
+			// The console is run on `localhost` so we replace 127.0.0.1 with localhost.
+			if addr.Hostname() == "127.0.0.1" {
+				addr.Host = "localhost" + ":" + addr.Port()
```

Comment:
FWIW we shouldn't run on "localhost" as it can trigger Mac security popups.

Reply:
Yeah this one is interesting. I had to add this to solve a problem with running locally with my mac and it's working currently, but I def want to do whatever is correct here. Lemme try a few things here before merging, but this part of the code has been around for a while.